src/sagemaker/amazon/common.py

```diff
@@ -171,8 +172,14 @@ def write_spmatrix_to_sparse_tensor(file, array, labels=None):
         array:
         labels:
     """
-
-    if not issparse(array):
+    try:
+        import scipy
+    except ImportError as e:
+        logging.warning("scipy is removed from SageMaker Python SDK v2.")
```

Comment:
you can still use the warning from the previous error message; just update it to not say "urllib" or "local mode" 😂 

e.g. "scipy failed to import. Sparse matrix functions will be impaired or broken."